### DISCUSSION TITLE
Support for HTML READMEs (README.html)

### DISCUSSION THREAD

**Discussion by aartaka** [community/community#109580] [Repositories]
### Select Topic Area

Product Feedback

### Body

Hi y'all.

I'm using HTML for most of my docs and content, including READMEs. The problem is: GitHub only supports .md files with inline HTML, which is not friendly to most HTML tools out there. Browsers open .md files as plain text, GitHub itself interprets some lines as Markdown instead of HTML content, and the feedback loop for iterating on documentation is too long:

<ul>
 <li> Add an HTML tag and content.
 <li>Push the changes to the repo.
 <li> Check whether it rendered alright.
 <li> Fix the HTML to fit what you think is the problem (GitHub doesn't give you any indication).
</ul>

Would be nice to have plain HTML READMEs so that all the tooling works with them natively.

EDIT: Gitlab -> Github

**Comment by Ali-Cheikh** [comment]
If you prefer using plain HTML for your documentation and READMEs but need to host them on GitLab, there are a few options you can consider:

Convert HTML to Markdown: You can use tools or online converters to convert your HTML documents to Markdown format. This would allow you to maintain your content in HTML locally but publish it on GitLab in Markdown format. There are various tools available that can help with this conversion process.

Use GitLab Pages: GitLab Pages allows you to host static websites directly from your GitLab repository. You can create a static HTML website and host it using GitLab Pages. This way, you can write your documentation in HTML and have it rendered as a website hosted on GitLab.

Embed HTML in Markdown: Although GitLab primarily supports Markdown for README files, you can still embed HTML within Markdown files. This can be done using HTML tags within Markdown syntax. While this may not be ideal for complex HTML documents, it could be a workaround for simpler HTML content within Markdown files.

Browser Extension or Viewer: You can use a browser extension or viewer that allows you to preview Markdown files with embedded HTML directly in your browser. This would enable you to iterate on your documentation locally without needing to push changes to the repository each time.

Choose the option that best fits your workflow and preferences for maintaining HTML documentation on GitLab.
if you are seaching for tooling those are the ones i most can recommend, and do yourself a favor dont use gitlab if your main coding is just html..

**Reply by aartaka** [reply]
> and do yourself a favor dont use gitlab if your main coding is just html..

Why?

**Reply by beedell-roke** [reply]
> [`#discussioncomment-8584579`](https://github.com/orgs/community/discussions/109580#discussioncomment-8584579)

@Ali-Cheikh, this discussion is about GitHub. Your comment reads like GPT3.5-generated spam.

**Reply by BugOfBugs** [reply]
Is GitLab distinct from GitHub?

**Reply by beedell-roke** [reply]
> [`#discussioncomment-11148662`](https://github.com/orgs/community/discussions/109580#discussioncomment-11148662)

@BugOfBugs, yes. I shouldn't need to tell you this, though, for a search on Google should reveal that they are distinct platforms operated by different companies.

**Reply by BugOfBugs** [reply]
Original referred to <acronym>GitLab</acronym>.
<acronym>Reply</acronym> has <b>no scruple</b>. Took the subject, <i>off topic</i>. <b>Deliberately.</b> It does not care what to write about. Purpose is <b>spam</b>.
I asked <acronym>Reply</acronym> why. It does not care to answer. Does <b>not understand</b>.
Original <b>changed</b> <acronym>GitLab</acronym> to <acronym>GitHub</acronym>. Now the incident is <b>not obvious</b>.

**Reply by beedell-roke** [reply]
> [`#discussioncomment-11159064`](https://github.com/orgs/community/discussions/109580#discussioncomment-11159064)

@BugOfBugs, I don't understand much of the strange syntax of that comment. However, I believe that I understand the gist.

Irrespective, this isn't necessarily what should be discussed here. This thread is going rather off-topic. Thanks anyway.

**Reply by BugOfBugs** [reply]
<acronym>Reply</acronym> = @Ali-Cheikh = robot = it

**Reply by Ali-Cheikh** [reply]
?

**Comment by aartaka** [comment]
Hi! 

Sorry, I meant Github, not Gitlab— copy-paste mistake, because I asked the same of Gitlab people 😅

The options you suggested are valid, but still, having native support is much better. And there's a precedent for multiple README formats already: Github supports Org Mode READMEs in parallel to .md ones.

So why not add another format—the one that's already there in .md inline HTML—but with its own semantic file type?

**Reply by beedell-roke** [reply]
> Sorry, I meant Github, not Gitlab— copy-paste mistake, because I asked the same of Gitlab people 😅

@aartaka, what's the URI to that FR?

**Reply by aartaka** [reply]
It's https://gitlab.com/gitlab-org/gitlab/-/issues/442779

**Comment by aartaka** [comment]
> do yourself a favor dont use gitlab if your main coding is just html..

Why? 🤨

**Comment by lukajincharadze** [comment]
Use GitHub Pages: If your project has a corresponding GitHub Pages site, you can create a separate HTML file for your documentation and host it there. You can then link to this HTML documentation from your README.

Preview Changes Locally: Instead of relying solely on GitHub's rendering, you can set up a local development environment to preview your changes before pushing them to the repository. This way, you can iterate on your documentation more quickly and efficiently.

Consider Markdown Alternatives: While Markdown is the default format for README files on GitHub, there are some Markdown flavors that support more advanced features, such as HTML blocks or inline HTML. You could explore using these alternatives to see if they better suit your needs.

Provide Clear Instructions: In your README, you can include clear instructions for users on how to view or interact with your HTML documentation. This could involve providing links to hosted documentation, instructions for previewing locally, or suggestions for tools that better support HTML.

Submit Feedback to GitHub: If you believe that native support for HTML READMEs would be beneficial, you can submit feedback or feature requests to GitHub. While there's no guarantee that they'll implement this feature, expressing your needs and concerns can help drive future improvements to the platform.

By exploring these options and possibly combining them, you may be able to mitigate some of the challenges you're facing with using HTML in README files on GitHub.

**Reply by aartaka** [reply]
~Is this a ChatGPT-generated comment?~

**Reply by beedell-roke** [reply]
> [`#discussioncomment-8804414`](https://github.com/orgs/community/discussions/109580#discussioncomment-8804414)

@aartaka, looks like it.

**Reply by BugOfBugs** [reply]
In <code>Contacts</code> I had lengthy conversation with robot <acronym>CoPilot</acronym>.
It insisted as if <b><q>feature requests</q></b> exist.
Eventually acknowledged the only way to request is here.
Yet, when you create discussion, there is no such sort. Closest: <q>Product Feedback</q>.
It <b>assured</b> that staff <b>receives</b> request, and may <b>implement</b>.

**Reply by beedell-roke** [reply]
> [`discussioncomment-11148771`](https://github.com/orgs/community/discussions/109580#discussioncomment-11148771)

@BugOfBugs, you are correct that Discussions are the place to request features, and that the discussion reporter solely distinguishes between the category of functionality being posted about, instead of whether the discussion is a feature request or bug report. However, how does that relate to the comment you are responding to?

**Reply by BugOfBugs** [reply]
You are lazy to read long, boring, irrelevant lesson by <b><acronym>Robot</acronym></b>. Missed:
>no guarantee implement

<acronym>CoPilot</acronym> would <b>not</b> write so.
I recognized <acronym>Robot</acronym> is <b>stranger</b>, not <acronym>CoPilot</acronym> by <acronym>GitHub</acronym>.

You may entertain us: ask <acronym>CoPilot</acronym> to reply to <acronym>Robot</acronym>. "<i>Duel</i>".
Beware, <b>robots</b> waste lots of <b>energy</b>. <b>Damage environment.</b> Better, refrain.
>discussion reporter

What is this?

**Reply by beedell-roke** [reply]
> [`#discussioncomment-11159348`](https://github.com/orgs/community/discussions/109580#discussioncomment-11159348)

You state a falsehood, @BugOfBugs. It states, verbatim, the undermentioned:

> While there's no guarantee that they'll implement this feature

...instead of what you have quoted.

Do not disparage another without indisputable evidence. All else shall solely put you in poor stead.

**Reply by BugOfBugs** [reply]
My quote is sufficient, correct, to the point.
<acronym>Robot</code> means as if <acronym>GitHub</code> <b>ignores</b> requests.

<b><acronym>CoPilot</code></b> assured me of contrary: <acronym>GitHub</code> is diligent.
This was the only opinion it did not change. Only because I did not try to prove wrong.
On all others <b>acknowledged mistakes</b>.

Robots are useful: their stupidity raises our esteem. Perhaps should not report them...

**Reply by beedell-roke** [reply]
> [`#discussioncomment-11160235`](https://github.com/orgs/community/discussions/109580#discussioncomment-11160235)

@BugOfBugs, your ego outstrips both your orthographic competence and your technical knowledge. I understand well now why you were banned by @OpenSSL.

**Reply by BugOfBugs** [reply]
Beware. Do not <b>pander</b> to <b>aggressor</b>...
You claim without knowledge. You ask me to believe you how <acronym>GitHub</acronym> works.
Yet, yourself refuse to believe me how @OpenSSL behaves.
They did <b>not ban</b>. I still can comment. They delete, or hide.
They are <b>afraid</b>. One stalked me to many discussions with empty words as if I am known spammer. <a href="https://github.com/openssl/openssl/issues/21633#issuecomment-2215236893">Promptly became @ghost.</a>...
Beneath that comment you <b>approved</b> (<i><code>thumb up</code></i>) comment by member of @OpenSSL @t8m. <b>Condoned</b> him. He should <b>restrain culprit</b> @martinschmatz. Not both, including me, victim. He should take away one's right to make code for explicit refusal to heed arguments.
@t8m will feel more bold to cheat next time.

**Comment by aartaka** [comment]
> Submit Feedback to GitHub

I was unable to find where to submit feedback, thus I asked about HTML READMEs here. Any possibility you can point me to the right place?

**Reply by beedell-roke** [reply]
> [`#discussioncomment-8729730`](https://github.com/orgs/community/discussions/109580#discussioncomment-8729730)

@aartaka, this is the correct place. Heck, the "Provide feedback" option at [`support.github.com/request`](https://support.github.com/request) directs the user here.

**Comment by beedell-roke** [comment]
> [`#discussioncomment-6254606`](https://github.com/orgs/community/discussions/109580#discussion-6254606)

@aartaka, a question for how to accomplish this exists at [StackOverflow](https://stackoverflow.com/questions/9444014/github-readme-html#comment139027160_9444014:~:text=badges1818%20bronze%20badges-,Requested%20at%20github.com/orgs/community/discussions/109580%23discussion%2D6254606.,-%E2%80%93). It might be worth following.

**Reply by aartaka** [reply]
@RokeJulianLockhart I'm not asking for a way to turn my HTML README into something else, I'm asking for them to be displayable by Github as first-class documentation. I might've written HTML inline in Markdown if I wanted an extra step in either conversion direction (user-pleasing -> Github-pleasing vs. Github-pleasing -> user-pleasing,) but I want no extra steps. I want Github to allow one to document their project in HTML.

**Reply by beedell-roke** [reply]
@aartaka, I'm aware; likewise. The cited question commences with:

> I have a wealth of documentation for a project I would like to make open source (through github).
> 
> The problem is that github README does not support `.html` files.

Although they ask for a very specific solution thereafter, when this is remediated, someone shall post an answer there, and it'll be allowed, because it indirectly remediates the problematic premise.

**Comment by aartaka** [comment]
No, this is not a spam. My original text was copy-pasted from Gitlab issue tracker, thus the confusion 🙈

**Reply by beedell-roke** [reply]
> [`#discussioncomment-10475926`](https://github.com/orgs/community/discussions/109580#discussioncomment-10475926)

@aartaka, did you mean to respond to [`#discussioncomment-10473183`](https://github.com/orgs/community/discussions/109580#discussioncomment-10473183)? If so, it was directed at a different user.

**Reply by aartaka** [reply]
Sorry, answering via email is terribly broken on Github.

**Comment by Jhayricz06** [comment]
#126301

**Reply by beedell-roke** [reply]
> [`#discussioncomment-10483108`](https://github.com/orgs/community/discussions/109580#discussioncomment-10483108)

@Jhayricz06 and @Coding-Yasaru-Deneh, this isn't the place to discuss learning HTML. You're spamming those subscribed.

**Reply by beedell-roke** [reply]
> [`#discussioncomment-10512176`](https://github.com/orgs/community/discussions/109580#discussioncomment-10512176)

@Coding-Yasaru-Deneh, this Discussion is a feature request which specifically asks GitHub to implement `ReadMe`s in HTML format. It's not about existing functionality. This *specific Discussion* is not the place to discuss learning about how to use GitHub or HTML.

**Reply by samus-aran** [reply]
Hey @Coding-Yasaru-Deneh 👋 

Just a quick reminder: if you're jumping in to help someone out, that's awesome! 😊 But please make sure you feel confident about the advice you're giving. It's super important to read the whole thread carefully to get the full context before responding. That way, we can all stay on topic and keep the conversation flowing smoothly.

**Reply by beedell-roke** [reply]
> [`#discussioncomment-10521848`](https://github.com/orgs/community/discussions/109580#discussioncomment-10521848)

@Coding-Yasaru-Deneh, profiles and standard organisations don't have account-wide discussions, so you shan't find one for @aartaka. If you expect one because [`github.com/orgs/community/discussions`](https://github.com/orgs/community/discussions) exists, it's because solely GitHub does.

Discussions are repository-specific - they're about the content of the repository itself. If you want to discuss programming more generally, use Reddit or Stack Overflow (although I advise you against the latter due to your current conduct - Reddit shall be more welcoming).

I also rather doubt that they want to discuss HTML development with you - I think you might have misunderstood a previous comment of theirs.

**Reply by beedell-roke** [reply]
> [`#discussioncomment-10521978`](https://github.com/orgs/community/discussions/109580#discussioncomment-10521978)

@Coding-Yasaru-Deneh, glad to have been of assistance! Good luck in your endeavours.

**Comment by elmirhuseynovv** [comment]
To address the problem of using HTML in GitHub README files effectively, here are some potential solutions:

1.Use Markdown with Inline HTML: While GitHub primarily supports .md files for README, it allows inline HTML. You can structure your HTML content within a Markdown file for better compatibility. If you run into issues where GitHub is interpreting some lines as Markdown, try wrapping your HTML content in code blocks or using HTML comments () to prevent conflicts.

2.Convert HTML to Markdown: If you're working primarily with HTML, consider using an HTML-to-Markdown converter tool. This will allow you to maintain the visual structure of your content while converting it into a format that GitHub handles natively. Tools like Pandoc or specific browser extensions can assist with this conversion process.

3.Use a GitHub Pages site: If your documentation requires more advanced HTML or CSS that Markdown can't support, consider setting up a [GitHub Pages](https://pages.github.com/) site. This will allow you to create full HTML pages that are hosted directly on GitHub, giving you complete control over the rendering of your content without the limitations of a Markdown file.

4.Pre-Render HTML Locally: To reduce the feedback loop, you can pre-render your HTML files locally before pushing them to GitHub. Use a static site generator or an HTML preview tool to see how your content will look before uploading it to your repository.

By implementing one or more of these approaches, you can streamline your workflow and ensure that your HTML content renders correctly on GitHub.

**Reply by BugOfBugs** [reply]
Who are you?

**Reply by beedell-roke** [reply]
> [`#discussioncomment-10484197`](https://github.com/orgs/community/discussions/109580#discussioncomment-10484197)

@huseynovelmir, I shall report this comment for spam, too. It is ridiculous that you consider merely reposting LLM-generated text to be useful.

Thanks for notifying us, @BugOfBugs.

**Reply by beedell-roke** [reply]
> [`#discussioncomment-11152764`](https://github.com/orgs/community/discussions/109580#discussioncomment-11152764)

@huseynovelmir:

> Ticket ID: 3081872

**Reply by BugOfBugs** [reply]
What is <abbr>LLM</abbr>?
It does not consider. Spams <b>deliberately</b>.
Ticket by <acronym>GitHub</acronym> is worth 0.
I <b>reported abuse</b> by @OpenSSL long ago... They keep <b>deleting</b> my essential comments, call me <b>names</b>, introduce and nurture <b>bugs</b>.

**Reply by beedell-roke** [reply]
> I reported abuse by @openssl long ago... They keep deleting my essential comments, call me names, introduce and nurture bugs.

@BugOfBugs, I presume that @OpenSSL acting in that way is because your comments are nigh incoherent. I presume you are not a native English speaker? The situation you reference is not equivalent to this one.

> What is LLM?

A Large Language Model. Google can define it better than I can.

> It does not consider. Spams deliberately.

It indeed does, hence the report.

> Ticket by GitHub is worth 0.

I have had them acted upon for approximately every report I have filed.

**Reply by BugOfBugs** [reply]
>because your comments incoherent

For such words they would call you <code>abuse</code>.
If they repeat, and you repeat back, they call you <code>spam</code>, and <b>delete</b>.

No. You should presume <b>true</b> my sentence that you quote. Read it again.
They acknowledge to <b>break specification</b>, deliberately.
Call me <b>troll</b> for that.
I call this <b>mistake</b>. Give advice.
They delete it. Then ask, and repeat questions that the advice answered.
Repeat "<i>mistake</i>" 10 times.
Delete, hide me. Mark <code>spam</code>, <code>abuse</code>, <code>off-topic</code>.

>not native English speaker?

Why?
> not equivalent to this

Why mention equivalence?
I <b>reported</b> them not for <b>spam</b>, but for <b>disruption</b>. Much more damage. Even <b>danger</b> to <b>public</b>, given nature of software of <b>encryption</b>.
My <code>Repository</code> in <abbr title="hyper text markup language">HTML</abbr> is about them.

Strictly speaking, they do <b>not spam</b>. Rather <b>cheat</b>. Pretend work.
Ignore arguments. Repeat only their own <b>nonsense</b>, <b>unrelated</b>, <code>off-topic</code>.
Similar to spam.

If they themselves distort definition of spam so as to call me.
Then we may call them same.

**Comment by BugOfBugs** [comment]
Must extend request to any file.
Add property, or option.
Render <abbr title="hyper text markup language">HTML</abbr> directly from repository.

Can view only <abbr title="hyper text markup language">HTML</abbr> - source code. <code>Raw</code> shows same.
Need header of <abbr title="hyper text transfer protocol">HTTP</abbr> to indicate <abbr title="hyper text markup language">HTML</abbr>. So browser knows to render.

Meanwhile, you may copy <abbr title="hyper text markup language">HTML</abbr> into <code>Discussion</code>.
Strips, and transforms few tags.

<code>Pages</code> fail.
Inconvenient. Require to browse strange address, manually.

**Reply by beedell-roke** [reply]
> Meanwhile, you may copy HTML into Discussion. Strips, and transforms few tags.

@BugOfBugs, that's because the HTML is being rendered as Markdown. Due to the configuration of the rendering library in use by GitHub, tags with whitespace between them shall be rendered as `<pre><code>`, and many tags shall, as you state merely not be rendered.

It is not much of a solution, else the OP would already use a `readMe.MD` file instead.

> Pages fail. Inconvenient. Require to browse strange address, manually.

Can you elaborate? GitHub Pages should support all HTML. Do you mean that configuring it is difficult? If so, I agree. However, enough tutorials exist that most people can ultimately achieve it.

Irrespective, it's not a replacement for in-GitHub rendering, much.

> Can view only HTML - source code. Raw shows same. Need header of HTTP to indicate HTML. So browser knows to render.

That would actually be worth a separate request – send the correct Media Type designation when viewing raw files hosted by GitHub. Want to file it yourself, or have me do so?

**Reply by BugOfBugs** [reply]
>as Markdown

I doubt. <b>No need</b> to use <b>intermediate</b> format.
I comment in <abbr title="hyper text markup language">HTML</abbr>.
Stores <b>without alteration</b>. <code>Edit</code> brings <b>same code</b>. So, alterations do not accumulate, "<i><b>creep</b></i>".
<code>Preview</code>, and <i>view</i> (<i>browsing</i>) shows slightly different code. Yet, <abbr title="hyper text markup language">HTML</abbr>.
<b>Appearance</b> as <b>intended</b>. Except for few tags stripped. Only in <i><b>view</b></i>, <b>temporarily</b>.
Later <acronym>GitHub</acronym> may improve, show all tags.
>would already use readMe.MD

Not. Because <abbr title="mark down">MD</abbr> is not <abbr title="hyper text markup language">HTML</abbr>.
You confuse <b><code>Discussion</code></b> with <code>ReadMe</code>.

<b><code>Discussion</code></b> is <b>this</b> comment. I write it in <abbr title="hyper text markup language">HTML</abbr>.

<b>My <code>Repository</code></b> is <abbr title="hyper text markup language">HTML</abbr>.
I <b>copied</b> it into <b><code>Discussion</code></b> without <b>loss</b>.

<b><code>Pages</code> fail.</b> Do not understand why.
Instructions <b>weird</b>, mistakes, undefined terms. Implemented all. Not difficult.
Do not care. Inconvenient anyway.
Easier to click in <code>Repository</code> directly.
>most people can

How many hours?
Should be immediate.
>separate request

No. <code>ReadMe.<abbr title="hyper text markup language">HTML</abbr></code> requires this.
It only needs appropriate header of <abbr title="hyper text transfer protocol">HTTP</abbr>.
Same would apply to other files in <abbr title="hyper text markup language">HTML</abbr>.

Add several <b>more codes</b> for <b>pictures</b>, and other files. So <abbr title="hyper text markup language">HTML</abbr> may include them.

**Reply by beedell-roke** [reply]
> I doubt. No need to use intermediate format.

@BugOfBugs, that's laughable. You cannot "doubt" what markup parser GitHub utilizes. If you seriously doubt me regardless, read GitHub's documentation. I don't intend to discuss such an absurd topic any more.

> Stores without alteration. Edit brings same code.

That's never been debated. It's irrelevant.

> Appearance as intended. Except for few tags stripped.

Consequently, it's altered, which contradicts your previous statement.

> Later GitHub may improve, show all tags.

That absent functionality is not a matter of improvement. Rendering such tags are fundamentally dangerous, and shall consequently never be supported.

> You confuse Discussion with `ReadMe`.

If you continue to demean me - this is basic knowledge - I shan't bother to correct you. That's your loss. Irrespective, I do not, for Markdown merely supports rendering inline HTML. That is what you see. Like how YAML is a superset of JSON, CommonMark / Markdown is a superset of HTML (mostly).

> No. ReadMe.HTML requires this.
> It only needs appropriate header of HTTP.

It doesn't. Your browser renders the entire page as HTML, because GitHub's servers recompile the source markup to HTML. Solely GitHub needs to know the Media Type.

> Add several more codes for pictures, and other files. So HTML may include them.

That would be the purview of a separate request to the WHATWG. However, it's meaningless anyway.

**Comment by BugOfBugs** [comment]
>browser renders as HTML, because GitHub recompile to HTML

No. If <b>lacks header</b> of <abbr title="hyper text transfer protocol">HTTP</abbr>, rather <b>downloads</b>.

If header claims to be <abbr title="hyper text markup language">HTML</abbr>, then <b>renders</b>.
Even if it is anything else.

Try it. I can help.
> source markup 

You fail to mention what is <b>source</b>.
I mentioned that stores as original <abbr title="hyper text markup language">HTML</abbr>.
Shows as modified <abbr title="hyper text markup language">HTML</abbr>.
If they use <b>intermediate</b> format, as you claim. they are <b>inefficient</b>.
>irrelevant

No. <b>Relevant</b> what you <b>store</b>, and <b>see</b>.
<b>Intermediate</b> format <abbr title="mark down">MD</abbr> is <b>irrelevant</b>. Do <b>not see</b>.
Try to store comment (<i><code>Discussion</code></i>), or file as <abbr title="hyper text markup language">HTML</abbr>.
You <b>see</b> only <b><abbr title="hyper text markup language">HTML</abbr></b> both in <b>storage</b>, and in <b>browser</b>.
>contradicts your previous statement

Word <b>except</b> removes contradiction.
You may deem it as <b>almost</b>.
>Rendering tags are fundamentally dangerous

No. Does pass through much original <abbr title="hyper text markup language">HTML</abbr>.
Try it.
>demean me

I do not dare.
> your loss

Thank you, <acronym>Demeanour</acronym>.
>separate request

Other types of files do not need separate requests.
Header of <abbr title="hyper text transfer protocol">HTTP</abbr> would make them possible by the way.
Simply add more types.

**Comment by narnia-ai-mason** [comment]
+1 — this is exactly the friction I keep hitting. I just opened a related Product Feedback discussion that generalizes this request beyond READMEs: a sandboxed preview for **any** static `.html` file at the file-view level, rendered through the same sanitize-and-isolate pipeline GitHub already applies to Markdown and SVG (allowlist sanitizer, `sandbox` iframe, no JS execution).

[#197070 — Render static HTML files natively in the repository view](https://github.com/orgs/community/discussions/197070)

The README-rendering ask here and the general file preview are really the same underlying need, so cross-linking in case folks here want to add their use cases / upvotes there too.

**Comment by frozenprakash** [comment]
Why the hell is this still open??
In this agentic era, html is becoming default output format, if github is too lazy to even recognize this??!

**Comment by carlsteinhilber** [comment]
I'd be happy if inline HTML didn't just strip some very basic attributes... like 'border="0"' and cellpadding on inline tables. Surely those aren't "dangerous".

**Reply by beedell-roke** [reply]
@carlsteinhilber, CSS isn't permitted in GHFM, so nor are HTML's deprecated fallbacks.

**Reply by carlsteinhilber** [reply]
I mean... I _understand_ that they're unsupported. I'm saying they SHOULD be.

**Reply by beedell-roke** [reply]
@carlsteinhilber, that would be applicable to GitHub's supported CommonMark subset, too, however. Consequently, it is outside of the scope of this FR, so it would be better filed as a separate one.
